config/decoder.go

```diff
@@ -342,7 +347,8 @@ func (d *decoder) mapping(childKey string, value reflect.Value, def string) erro
 		for key := range v {
 			subKey := fmt.Sprintf("%v", key)
 			if subKey == "" {
-				return fmt.Errorf("empty key leads to ambiguity for path: %q", childKey)
+				// We can confuse an empty map key with a root element.
+				return addKeyToError(childKey, errors.New("empty map key is ambigious"))
```

Comment:
I would rather to keep an error here, so stacktrace is not going to get too long.